### DISCUSSION TITLE
SYS: cfAssertNormalStart [FAIL]

### DISCUSSION THREAD

**Discussion by alexandraH99** [bitcraze/discussions#103] [Q&A]
Hi!

Here is what I saw in the console after I modified the firmware:
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.05
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: The system resumed after a failed assert [WARNING]
SYS: Assert failed at ../crazyflie-firmware/src/utils/interface/cf_math.h:61
SYS: cfAssertNormalStart [FAIL]

Does anyone have any idea of what causes this error?

A related question might be: I wrote a controller basically just makes the the crazyflie hover in the air. How can I activate the crazyflie to hover and stop? Like what should I do so that once I start the Crazyflie, it starts to hover after a few seconds, and then it can stop with some command? 

Thanks!

**Comment by knmcguire** [comment]
Hi!

Questions about your first issue: 
- Which firmware version have you flashed (latest one that branched off from before your modifications)
- Are you using any function cf_math somewhere in your latest additions?

For your second question: please make a new discussion for this. We prefer one topic per discussion: https://github.com/orgs/bitcraze/discussions/55

**Reply by alexandraH99** [reply]
Hi!

- I'm not entirely sure, I basically just cloned from the repository a few days ago, made my modifications, and then flashed.
- Yes I'm using the matrix multiplication and transpose function, and no errors related to these functions showed up during compilation.

Thanks!

**Comment by krichardsson** [comment]
The problem is that your matrix data is not aligned on a 4 byte boundary, which is required by the matrix functions.
